cmd/prepare-storage.go

```diff
@@ -69,21 +78,40 @@ func formatXLMigrateLocalEndpoints(endpoints EndpointList) error {
 
 // Cleans up tmp directory of local disks.
 func formatXLCleanupTmpLocalEndpoints(endpoints EndpointList) error {
-	for _, endpoint := range endpoints {
+	g := errgroup.WithNErrs(len(endpoints))
+	for index, endpoint := range endpoints {
 		if !endpoint.IsLocal {
 			continue
 		}
-		formatPath := pathJoin(endpoint.Path, minioMetaBucket, formatConfigFile)
-		if _, err := os.Stat(formatPath); err != nil {
-			if os.IsNotExist(err) {
-				continue
+		index := index
+		g.Go(func() error {
+			epPath := endpoints[index].Path
+			formatPath := pathJoin(epPath, minioMetaBucket, formatConfigFile)
+			if _, err := os.Stat(formatPath); err != nil {
+				if os.IsNotExist(err) {
+					return nil
+				}
+				return err
 			}
-			return err
-		}
-		if err := removeAll(pathJoin(endpoint.Path, minioMetaTmpBucket)); err != nil {
-			return err
-		}
-		if err := mkdirAll(pathJoin(endpoint.Path, minioMetaTmpBucket), 0777); err != nil {
+			if _, err := os.Stat(pathJoin(epPath, minioMetaTmpBucket+"-old")); err != nil {
+				if !os.IsNotExist(err) {
+					return err
+				}
+			}
+
+			if err := renameAll(pathJoin(epPath, minioMetaTmpBucket),
```

Comment:
A comment describing the approach would be useful. It is not obvious that we need to a create unique directory on every startup to move temporary objects left behind from previous runs of minio server.
Suggestion:
```go
// Need to move temporary objects left behind from previous run of minio server 
// to a unique directory under `minioMetaTmpBucket-old` to clean up 
// `minioMetaTmpBucket` for the current run. 
// E.g,
// /disk1/.minio.sys/tmp-old/
// ├── 33a58b40-aecc-4c9f-a22f-ff17bfa33b62
// └── e870a2c1-d09c-450c-a69c-6eaa54a89b3e
//
// In this example, `33a58b40-aecc-4c9f-a22f-ff17bfa33b62` directory contains 
// temporary objects from one of the previous runs of minio server.
```

Reply:
Copy pasted